maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java

```diff
@@ -40,22 +41,39 @@
  *     <code>1.0alpha1 =&gt; [1, 0, alpha, 1]</code></li>
  * <li>unlimited number of version components,</li>
  * <li>version components in the text can be digits or strings,</li>
- * <li>strings are checked for well-known qualifiers and the qualifier ordering is used for version ordering.
- *     Well-known qualifiers (case insensitive) are:<ul>
- *     <li><code>alpha</code> or <code>a</code></li>
- *     <li><code>beta</code> or <code>b</code></li>
- *     <li><code>milestone</code> or <code>m</code></li>
- *     <li><code>rc</code> or <code>cr</code></li>
- *     <li><code>snapshot</code></li>
- *     <li><code>(the empty string)</code> or <code>ga</code> or <code>final</code></li>
- *     <li><code>sp</code></li>
- *     </ul>
- *     Unknown qualifiers are considered after known qualifiers, with lexical order (always case insensitive),
- *   </li>
- * <li>a hyphen usually precedes a qualifier, and is always less important than something preceded with a dot.</li>
+ * <li>
+ *   String qualifiers are ordered lexically (case insensitive), with the following exceptions:
+ *   <ul>
+ *     <li> 'snapshot' &lt; '' &lt; 'sp' </li>
+ *   </ul>
+ *   and alias -&gt; replacement (all case insensitive):
+ *   <ul>
+ *     <li> 'a' -&gt; 'alpha' </li>
+ *     <li> 'b' -&gt; 'beta' </li>
+ *     <li> 'm' -&gt; 'milestone' </li>
+ *     <li> 'cr' -&gt; 'rc' </li>
+ *     <li> 'final' -&gt; '' </li>
+ *     <li> 'final' -&gt; '' </li>
+ *     <li> 'final' -&gt; '' </li>
+ *   </ul>
```

Comment:
Please retain the formatting with `<code/>` to retain readability.

Reply:
will do

Reply:
Does this format really add a benefit here, except for the exception documentation?

Reply:
trying to show and explain why there is hard coded qualifiers. the current version took the form it has following change requests from elharo. any other form is good to me.

Reply:
<li><code> alpha = a &lt; beta = b &lt; milestone = m &lt; rc = cr
 *         &lt; snapshot &lt; '' = final = ga = release &lt; sp </code></li>

Reply:
changed to a simpler form